lib/class-factory.js

```diff
@@ -177,13 +182,36 @@ function ClassFactory (vm) {
           }
         };
 
-        C = ensureClass(getClassHandle, className);
+        try {
+          C = ensureClass(getClassHandle, className);
+        } finally {
+          setUsedClass(className, C);
+        }
       }
     }
 
     return new C(null);
   };
 
+  function getUsedClass(className) {
+    let kclass;
```

Comment:
```suggestion
    let klass;
```

^ Perhaps we should name it `klass` for consistency?